hack/cluster-openshift.sh

```diff
@@ -541,19 +547,42 @@ if [ "$_CMD" = "up" ]; then
     fi
   fi
 
+  checkApp () {
+    local expected="$1"
+    apps=$(${MAISTRA_ISTIO_OC_COMMAND} get deployment.apps -n istio-system -o jsonpath='{range .items[*]}{.metadata.name}{" "}{end}' 2> /dev/null)
+    for app in ${apps[@]}
+    do
+	  if [[ "$expected" == "$app" ]]; then
+	    return 0
+	  fi
+    done
+    return 1
+  }
+
   # If Istio is enabled, it should be installing now - if we need to, wait for it to finish
   if [ "${ISTIO_ENABLED}" == "true" ] ; then
     if [ "${KIALI_ENABLED}" == "true" -o "${WAIT_FOR_ISTIO}" == "true" ]; then
-      echo "Wait for Istio to fully start (this is going to take a while)..."
+      echo -n "Wait for Istio to fully start (this is going to take a while)..."
 
-      echo -n "Waiting for Istio Deployments to be created..."
-      while [ "$(${MAISTRA_ISTIO_OC_COMMAND} get pods -l job-name=openshift-ansible-istio-installer-job -n istio-system -o jsonpath='{.items..status.conditions[0].reason}' 2> /dev/null)" != "PodCompleted" ]
+      ## Check if jaeger is part of the expected apps
+      if [ "${TRACING_ENABLED}" == "false" ]; then
+        expected_apps=(grafana istio-citadel istio-egressgateway istio-galley istio-ingressgateway istio-pilot istio-policy istio-sidecar-injector istio-telemetry prometheus)
+      else
+        expected_apps=(grafana istio-citadel istio-egressgateway istio-galley istio-ingressgateway istio-pilot istio-policy istio-sidecar-injector istio-telemetry jaeger-collector jaeger-query prometheus)
```

Comment:
put grafana at the end of the list since that is created by Maistra after all the others have started.

For tracing, I believe there is also an elasticsearch pod we should be waiting for. Is that correct? If so, we should add that to the list since it is part of the jaeger install.